platform/windows/display_server_windows.cpp

```diff
@@ -2187,8 +2196,47 @@ LRESULT DisplayServerWindows::MouseProc(int code, WPARAM wParam, LPARAM lParam)
 	return ::CallNextHookEx(mouse_monitor, code, wParam, lParam);
 }
 
-// Our default window procedure to handle processing of window-related system messages/events.
-// Also known as DefProc or DefWindowProc.
+// handle windows message received while CreateWindowEx is still on the stack and our data
+// structures are not fully initialized
+LRESULT DisplayServerWindows::_handle_early_window_message(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam) {
+	switch (uMsg) {
+		case WM_GETMINMAXINFO:
+			// we receive this during CreateWindowEx and we haven't initialized the window
+			// struct, so let Windows figure out the maximized size
+			// silently forward to user/default
+			break;
+		case WM_NCCREATE: {
+			// we tunnel an unowned pointer to our window context (WindowData) through the
+			// first possible message (WM_NCCREATE) to fix up our window context collection
+			CREATESTRUCTW *pCreate = (CREATESTRUCTW *)lParam;
+			WindowData *pWindowData = reinterpret_cast<WindowData *>(pCreate->lpCreateParams);
+
+			// fix this up so we can recognize the remaining messages
+			pWindowData->hWnd = hWnd;
+			break;
+		}
+		default: {
+			// additional messages during window creation should happen after we fixed
+			// up the data structures on WM_NCCREATE, but this might change in the future,
+			// so report an error here and then we can implement them
+			bool error;
+			Array args;
+			args.push_back(uMsg);
+			String message = String("unexpected windows message 0x%x received for window we cannot recognize in our collection; sequence error").sprintf(args, &error);
+			if (!error) {
+				ERR_PRINT_ONCE(message);
+			}
```

Comment:
yes sorry I hadn't found vformat the wrapper :) sorry to waste your time on that

Reply:
am I supposed to accept your suggestion (doesn't that create an extra commit?) or should I amend it and force push?

Reply:
Made the change and amended commit.  I wasn't sure if I am supposed to accept suggestions, because that creates more commits?  Decided to go with what is the least work for others.